wolfssl/wolfcrypt/aes.h

```diff
@@ -394,6 +399,27 @@ typedef struct XtsAes {
 } XtsAes;
 #endif
 
+#if (!defined(WC_AESFREE_IS_MANDATORY)) &&                              \
```

Comment:
Would it be simpler to check this in `configure.ac`? E.g. just forbid the combination of `--enable-afalg` with opensslall or extra?

Reply:
Most things that can be done in the headers should be, so that non-autotools use cases are covered.  And that's certainly the case here, because there are many non-autotools users of `OPENSSL_EXTRA`, and they're directly impacted by the issue.

Reply:
Oh and @philljj I forgot to note, the combination isn't forbidden -- `AFALG` _can_ be combined with `OPENSSL_EXTRA` and that's a useful combination.  It's just that those several legacy low-level crypto APIs have to be masked out.  There may be other low level APIs with the same issue btw, but I confined this effort to AES.  For comparison, EVP very much has cleanup APIs, and EVP is usefully brought in by `OPENSSL_EXTRA`.